app/src/main/java/org/mozilla/focus/activity/MainActivity.kt

```diff
@@ -112,6 +128,45 @@ open class MainActivity : LocaleAwareAppCompatActivity() {
         AppReviewUtils.showAppReview(this)
     }
 
+    private fun setSplashScreenPreDrawListener() {
+        val content: View = findViewById(android.R.id.content)
+        content.viewTreeObserver.addOnPreDrawListener(object : ViewTreeObserver.OnPreDrawListener {
+            override fun onPreDraw(): Boolean {
+                experimentsRequestExpired = settings.experimentsStartFetchingTimestamp?.let { startTimestamp ->
+                    System.currentTimeMillis() - startTimestamp >= REQUEST_TIME_OUT
+                } ?: false
+
+                return if (experimentsRequestExpired || components.settings.experimentsApplied) {
```

Comment:
The logic here is pretty convoluted. Quite a lot of state being tracked, but without the benefit. 

Given you don't seem to be dismissing the splashscreen early, this could be more easily implemented as 

1. show the splash screen
2. `fetchExperiments()`
3. wait `REQUEST_TIME_OUT`
4. `applyPendingExperiments()`
5. dismiss the splash screen.

For step 4, we can call `applyPendingExperiments()` even if the fetch hasn't finished.

This way we don't need to do anything with a `Nimbus.Observer`, much less, keep track of whether or not we've called `applyPendingExperiments()`.